typescript/web/src/components/labeling-tool/openlayers-map/labels.tsx

```diff
@@ -54,7 +60,11 @@ export const Labels = ({
     variables: { imageId: imageId as string },
   });
   const selectedLabelId = useLabelingStore((state) => state.selectedLabelId);
+  const selectedTool = useLabelingStore((state) => state.selectedTool);
   const labels = data?.image?.labels ?? previousData?.image?.labels ?? [];
+  const selectedLabel = labels.filter(
```

Comment:
use `array.find()` instead of `array.filter()[0]` . It's faster because it stops iterating as soon as it finds it.